### PR TITLE
E2E tests CI bootstraps

### PR DESCRIPTION
**Pull request by shhsu**

### CODE REVIEW

File: NuGet.config [hunk 1]
```diff
@@ -2,8 +2,9 @@
 <configuration>
   <packageSources>
     <clear />
-    <add key="NuGet2" value="https://api.nuget.org/v3/index.json" />
+    <!-- <add key="Core" value="http://dotnet.myget.org/F/dotnet-core" /> -->
```
Comment (jkotalik) [inline NuGet.config]: Can you do the same thing we do with nuget config files: https://github.com/aspnet/IISIntegration/blob/dev/NuGet.config
and https://github.com/aspnet/IISIntegration/blob/dev/build/sources.props

File: build/sources.props [hunk 2]
```diff
@@ -1,16 +0,0 @@
-<Project>
-  <Import Project="$(DotNetRestoreSourcePropsPath)" Condition="'$(DotNetRestoreSourcePropsPath)' != ''"/>
-
-  <PropertyGroup Label="RestoreSources">
-    <RestoreSources>$(DotNetRestoreSources)</RestoreSources>
```
Comment (jkotalik) [inline build/sources.props:5]: See above and keep the sources.props file.

File: test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj [hunk 3]
```diff
@@ -20,11 +20,11 @@
...
 
   <ItemGroup>
-    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="$(MicrosoftAspNetCoreTestingPackageVersion)" />
-    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="$(MicrosoftNETTestSdkPackageVersion)" />
+    <PackageReference Include="Microsoft.AspNetCore.Testing" Version="2.1.0-preview1-27644" />
+    <PackageReference Include="Microsoft.NET.Test.Sdk" Version="15.3.0" />
     <PackageReference Include="System.Runtime.CompilerServices.Unsafe" Version="$(SystemRuntimeCompilerServicesUnsafePackageVersion)" />
-    <PackageReference Include="xunit" Version="$(XunitPackageVersion)" />
-    <PackageReference Include="xunit.runner.visualstudio" Version="$(XunitRunnerVisualstudioPackageVersion)" />
+    <PackageReference Include="xunit" Version="2.3.1" />
```
Comment (jkotalik) [inline test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj:26]: Define these versions in dependencies.props.

File: test/AspNetCoreModule.Test/Framework/TestWebSite.cs [hunk 4]
```diff
@@ -292,6 +292,24 @@ public TestWebSite(IISConfigUtility.AppPoolBitness appPoolBitness, string logger
...
                     }
 
+
+                    if (TestFlags.Enabled(TestFlags.UseDotNetCore22))
+                    {
+                        string aspNetCoreModuleHostingModel, buildConstants;
+
+                        if (TestFlags.Enabled(TestFlags.InprocessMode))
+                        {
+                            aspNetCoreModuleHostingModel = "inprocess";
```
Comment (jkotalik) [inline test/AspNetCoreModule.Test/Framework/TestWebSite.cs]: Why do you need two of these? Can't you reused the same one?

File: tools/versions.psm1 [hunk 5]
```diff
@@ -0,0 +1,7 @@
+
+function GetASPNETVersionUsedByBuild() {
+    $dependenciesPropFile = [System.IO.Path]::Combine($PSScriptRoot, "..", "build", "dependencies.props")
```
Comment (jkotalik) [inline tools/versions.psm1]: Comment about what this function does.

File: test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22 [hunk 6]
```diff
@@ -1,17 +1,16 @@
 ﻿<?xml version="1.0" encoding="UTF-8"?>
```
Comment (jkotalik) [inline test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22:1]: Why do we have 3 different csproj for for different dotnet version? You can specify multiple through adding <TargetFrameworks>netcoreapp2.0;netcoreapp2.1;netcoreapp2.2</TargetFrameworks>

**Review by jkotalik — CHANGES_REQUESTED** [review]

**Comment by jkotalik** [comment]
Can you take a look at the appveyor failures: https://ci.appveyor.com/project/aspnetci/aspnetcoremodule/build/1.0.260. Seems to be two sln files?

File: NuGet.config (hunk 1 above)
Comment (shhsu) [reply NuGet.config]: Including in next revision

File: build/sources.props (hunk 2 above)
Comment (shhsu) [reply build/sources.props:5]: including in next revision

File: test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj (hunk 3 above)
Comment (shhsu) [reply test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj:26]: Right now I can't allow korebuild to upgrade these pacakges. The tests don't seem to work with the newer version of xunit and .net test framework. The error is really vague and hard to debug so I am leaving them unchanged. At least for now.

File: test/AspNetCoreModule.Test/Framework/TestWebSite.cs (hunk 4 above)
Comment (shhsu) [reply test/AspNetCoreModule.Test/Framework/TestWebSite.cs]: It's the same file. `aspNetCoreModuleHostingModel` is a build parameter

File: tools/versions.psm1 (hunk 5 above)
Comment (shhsu) [reply tools/versions.psm1]: Including in the next revision

File: tools/install_ancm.ps1 [hunk 7]
```diff
@@ -82,9 +82,8 @@ function Invoke-UpdateAppHostConfig {
     ## Trick: if the worker process was running 32 bit modes, %windir%\System32 is actually redirected to %windir%\SysWOW64
     ## Hence we would always point to module to $inetSrvDirs.x64
     $aspNetCoreHandlerFilePath = Join-Path $inetSrvDirs.x64 "aspnetcorev2.dll"
-    Start-IISCommitDelay
-    $sm = Get-IISServerManager
 
+    $sm = Get-IISServerManager
```
Comment (jhkimnew) [inline tools/install_ancm.ps1]: Why did you change install_ancm.ps1? As far as I know we don't use this one.

**Review by shhsu — COMMENTED** [review]
talked offline. we will migrate to vsts

File: tools/install_ancm.ps1 (hunk 7 above)
Comment (shhsu) [reply tools/install_ancm.ps1]: We don't currently but vsts will be using it. I modified an existing script

File: test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22 (hunk 6 above)
Comment (shhsu) [reply test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22:1]: Talked offline. I can make a new PR to change this but it's a bit out of scope of this PR

File: tools/install_ancm.ps1 (hunk 7 above)
Comment (jhkimnew) [reply tools/install_ancm.ps1]: Considering this test for E2E, we should not use any custom script and use installer (msi) instead. In addition, using custom install script will cause only complexity. Please consider not to use this even for vsts. :-)

File: test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj (hunk 3 above)
Comment (jkotalik) [reply test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj:26]: File an issue for it then.

File: tools/install_ancm.ps1 (hunk 7 above)
Comment (jkotalik) [reply tools/install_ancm.ps1]: No msi; windows hosting bundle sure.

File: test/AspNetCoreModule.Test/Framework/TestWebSite.cs (hunk 4 above)
Comment (jkotalik) [reply test/AspNetCoreModule.Test/Framework/TestWebSite.cs]: Can you consolidate build parameters into one parameter?

File: tools/install_ancm.ps1 (hunk 7 above)
Comment (shhsu) [reply tools/install_ancm.ps1]: Talked to @jkotalik . Eventually we can use the windows hosting bundle msi to install both sharefx and ancm. Right now there's still work to be done to include ancmV2 in the hosting bundle. I will create a new work item for it.

Comment (shhsu) [reply tools/install_ancm.ps1]: https://github.com/aspnet/AspNetCoreModule/issues/315

File: test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj (hunk 3 above)
Comment (shhsu) [reply test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj]: https://github.com/aspnet/AspNetCoreModule/issues/315

Comment (shhsu) [reply test/AspNetCoreModule.Test/AspNetCoreModule.Test.csproj:26]: https://github.com/aspnet/AspNetCoreModule/issues/316

File: test/AspNetCoreModule.Test/Framework/TestWebSite.cs (hunk 4 above)
Comment (shhsu) [reply test/AspNetCoreModule.Test/Framework/TestWebSite.cs]: `aspNetCoreModuleHostingModel` is used here 
https://github.com/aspnet/AspNetCoreModule/blob/10637ae513fadd27c4dc33cc5d5bcdf2ab9e5d4c/test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22#L8
and consumed by IISIntegration
`BuildConstants` is used here
https://github.com/aspnet/AspNetCoreModule/blob/10637ae513fadd27c4dc33cc5d5bcdf2ab9e5d4c/test/AspNetCoreModule.TestSites.Standard/AspNetCoreModule.TestSites.Standard.csproj.UseDotNetCore22#L9
and used as a compiler directive to determine to call "UseIIS" or "UseIISIntegration"

Comment (shhsu) [reply test/AspNetCoreModule.Test/Framework/TestWebSite.cs]: Can't really separate them as they are used by different consumers

**Comment by shirhatti** [comment]
@shhsu Are you monitoring the VSTS failures? https://devdiv.visualstudio.com/DevDiv/Default/_build/results?buildId=1740832&view=logs
Doesn't look like this is passing yet

**Comment by shhsu** [comment]
Builds are not passing because of some configuration issues I am still investigating. Each component in here is tested locally

**Review by jkotalik — APPROVED** [review]
